src/components/input/inputNumber/inputNumber.test.tsx

```diff
@@ -0,0 +1,155 @@
+            testIncrementLogic({ expectedValue: '1' });
+        });
+
+        it('should return the maximum when the newly generated value exceeds the maximum', () => {
+            const max = 5;
+            const step = 2;
+            const value = '4';
+            testIncrementLogic({ max, step, value, expectedValue: max.toString() });
+        });
+
+        it('should increment by floating point value when the step is a float', () => {
+            const value = '1';
+            const step = 0.5;
+            testIncrementLogic({ step, value, expectedValue: (Number(value) + step).toString() });
+        });
+
+        it('should round down to the nearest multiple of the step before incrementing by the step value', () => {
+            const value = '1';
+            const step = 0.3;
+            testIncrementLogic({ step, value, expectedValue: '1.2' });
+        });
+
+        it('should increment to the minimum when no value is provided', () => {
+            const step = 6;
+            const min = 5;
+            const max = 10;
+            testIncrementLogic({ step, min, max, expectedValue: min.toString() });
+        });
+    });
+
+    describe('decrement button', () => {
+        const useNumberMaskMock = jest.spyOn(InputHooks, 'useNumberMask');
+
+        afterEach(() => {
+            useNumberMaskMock.mockReset();
+        });
+
+        const testDecrementLogic = ({
+            expectedValue,
+            ...props
+        }: Partial<IInputNumberProps> & { expectedValue: string }) => {
+            const setValue = jest.fn();
+            const hookResult = {
+                setValue,
+                value: props.value,
+                unmaskedValue: props.value,
+            } as unknown as InputHooks.IUseNumberMaskResult;
+            useNumberMaskMock.mockReturnValue(hookResult);
+
+            render(createTestComponent({ ...props }));
+
+            const [decrementButton] = screen.getAllByRole<HTMLButtonElement>('button');
+            fireEvent.click(decrementButton);
+
+            expect(setValue).toHaveBeenCalledWith(expectedValue);
+        };
+
+        it('should decrement by step', () => {
+            const value = '10';
+            const step = 2;
+            const expectedValue = (10 - 2).toString();
+            testDecrementLogic({ value, step, expectedValue });
+        });
+
+        it('should decrement to the minimum when no value provided', () => {
+            const step = 2;
+            const min = 1;
+            testDecrementLogic({ step, min, expectedValue: min.toString() });
+        });
+
+        it('should decrement to maximum if the initial value minus the step is greater than the maximum', () => {
+            const value = '100';
+            const step = 2;
+            const min = 1;
+            const max = 50;
+            testDecrementLogic({ value, step, min, max, expectedValue: max.toString() });
+        });
```

Comment:
We won't have the case where `value` is greater than `max`, right? I think there's a similar test where `value` is less than `min`. I'd suggest to remove these two tests to keep the tests aligned with the logic implemented on the component

Reply:
It's just this one. Less than `min` is when `value` is undefined. Good catch